ADAL/tests/unit/ADAuthenticationErrorTests.m

```diff
@@ -40,70 +40,88 @@ - (void)tearDown
+}
+
+- (void)testErrorFromArgument_whenArgumentNilArgumentNameNilCorrelationIdNil_shouldThrow
+{
     XCTAssertThrowsSpecificNamed([ADAuthenticationError errorFromArgument:nil argumentName:nil correlationId:nil],
                                  NSException, NSInvalidArgumentException, "Nil argument name should throw.");
 }
 
-- (void)testErrorFromArgumentNil
+- (void)testErrorFromArgument_whenArgumentNilArgumentNameIsValidCorrelationIdNil_shouldReturnError
 {
-    NSString* parameter = @"parameter123456 %@";
-    //nil value:
-    ADAuthenticationError* error = [ADAuthenticationError errorFromArgument:nil argumentName:parameter correlationId:nil];
-    XCTAssertNotNil(error, "No error for nil prameter");
-    [self adValidateForInvalidArgument:parameter error:error];
-    XCTAssertTrue([error.errorDetails containsString:@"(null)"], "'null' should be part of the text");
+    NSString *parameter = @"parameter123456 %@";
+    
+    ADAuthenticationError *error = [ADAuthenticationError errorFromArgument:nil argumentName:parameter correlationId:nil];
+    
+    XCTAssertNotNil(error);
+    ADAssertStringEquals(error.domain, ADAuthenticationErrorDomain);
+    XCTAssertNil(error.protocolCode);
+    ADAssertStringEquals(error.errorDetails, @"The argument 'parameter123456 %@' is invalid. Value:(null)");
 }
 
-- (void)testErrorFromArgumentNormal
+- (void)testErrorFromArgument_whenArgumentIsValidArgumentNameIsValidCorrelationIdNil_shouldReturnError
 {
-    NSString* parameter = @"parameter123456 %@";
-    NSString* parameterValue = @"value1245 %s@";
-    ADAuthenticationError* error = [ADAuthenticationError errorFromArgument:parameterValue argumentName:parameter correlationId:nil];
-    XCTAssertNotNil(error, "No error for valid prameter");
+    NSString *parameter = @"parameter123456 %@";
+    NSString *parameterValue = @"value1245 %s@";
     
-    [self adValidateForInvalidArgument:parameter error:error];
-    XCTAssertTrue([error.errorDetails containsString:parameterValue], "Value should be part of the text");
+    ADAuthenticationError *error = [ADAuthenticationError errorFromArgument:parameterValue argumentName:parameter correlationId:nil];
+    
+    XCTAssertNotNil(error);
+    ADAssertStringEquals(error.domain, ADAuthenticationErrorDomain);
+    XCTAssertNil(error.protocolCode);
+    ADAssertStringEquals(error.errorDetails, @"The argument 'parameter123456 %@' is invalid. Value:value1245 %s@");
 }
 
-- (void)testErrorFromOAuthError
+#pragma mark - errorFromAuthenticationError
+
+- (void)testErrorFromAuthenticationError_whenErrorOAuthHprotocolCodeValiderrorDetailsValidCorrelationIdNil_shouldReturnError
 {
-    NSString* details = @"Some details";
-    NSString* protocolCode = @"procol code";
-    ADAuthenticationError* error = [ADAuthenticationError errorFromAuthenticationError:AD_ERROR_SERVER_OAUTH protocolCode:protocolCode errorDetails:details correlationId:nil];
+    NSString *details = @"Some details";
```

Comment:
changed to this one: testErrorFromAuthenticationError_whenErrorValidCodeValidErrorDetailsValidCorrelationIdNil_shouldReturnErrorWithDescription Please let me know if you have better name :)